### ISSUE TITLE
SIGSEGV in TSymbolTableLevel::insert when parsing quote character directly after a macro name.

### ISSUE THREAD

**Issue by xorgy** [KhronosGroup/glslang#39]
```GLSL
#define A()
void a = A"
```

**Comment by johnkslang** [comment]
This is a good example of cascading errors, which if prevented by default, there would be no problem here.  Glslang correctly caught all these before having the issue:

```
ERROR: 0:2: 'string' : end of line in string
ERROR: 0:2: 'macro expansion' : expected '(' following A
ERROR: 0:2: '""' : string literals not supported
ERROR: 0:2: '' : undeclared identifier
ERROR: 0:2: 'a' : illegal use of type 'void'
```

It was easy to fix this one (91b977e17217eea47c2bd0568e52df9db0c3c29d), but perhaps energy is better spent on the scheme to kick out early when seeing the first error.

**Comment by xorgy** [comment]
I think maybe we can't really be sure that there ARE any errors after the first error.
When you fix the first error, it could affect all of the others.

I  think that theoretically no validator can tell you with certainty that there are any errors after it encounters the first, except maybe it can depend on what kind of errors occur.


I've seen validators fail after "too many errors", maybe one could decide whether an error affects other error cases(i.e. a syntax error in a declaration or definiton makes a symbol undeclared or undefined later, so you can't just continue parsing).
